Fixes #3515 - Review Uptime.

sbordet opened the pull request:
* Deprecated for removal Uptime.
* Moved ServerMBean.instant to Server.
* Removed Uptime usages, replaced by using Instant in Server.
* Removed Uptime usage in AbstractLifeCycle, only used for DEBUG logging.

sbordet committed:
Fixes #3515 - Review Uptime.

* Deprecated for removal Uptime.
* Remove its usages, replaced by using Instant in Server.

Signed-off-by: Simone Bordet <simone.bordet@gmail.com>

gregw reviewed (changes requested):
there are merge conflicts

sbordet committed:
Merge remote-tracking branch 'origin/jetty-12.1.x' into fix/jetty-12.1.x/3515-remove-uptime

# Conflicts:
#	jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/Server.java

sbordet committed:
Updated deprecation to Jetty 12.1.4.

Signed-off-by: Simone Bordet <simone.bordet@gmail.com>

joakime commented on jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/Server.java:
```diff
@@ -530,20 +532,36 @@ public HttpField getDateField()
+     */
+    public long getUptimeMillis()
+    {
+        return Duration.between(startupInstant, Instant.now()).toMillis();
```
Wouldn't this calculation be wrong twice a year during DST changes?

sbordet replied:
Yes, but in order to fix this we would need to know the `ZoneId`. 
We could use `ZoneId.systemDefault()`, and modernize `Server` to have a `ZonedDateTime` as startup "time", rather than just `Instant`.
If users want to configure a different timezone for the JVM, they can use the system property `user.timezone`.

joakime reviewed (dismissed):
Approved, but it seems like we would have an insignificant bug thru a DST change.

If so, would a negative value cause any bugs?

sbordet committed:
Updates from review.

Now the startupTime is a ZonedDateTime, which takes into account daylight savings.

Signed-off-by: Simone Bordet <simone.bordet@gmail.com>

lorban commented on jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/jmx/ServerMBean.java:
```diff
@@ -66,16 +62,16 @@ public List<ContextHandler> getContexts()
     }
 
-    @ManagedAttribute("The UTC startup instant")
+    @ManagedAttribute("The startup date time in the system timezone")
```
This is a somewhat breaking change, although not necessarily a bad one.

lorban reviewed (dismissed):
LGTM if we accept the tiny breakage.

sbordet committed:
Updates from review.

Signed-off-by: Simone Bordet <simone.bordet@gmail.com>
